relay/traffic/handler.go

```diff
@@ -137,7 +137,9 @@ func (handler *Handler) handleHttp(clientResponse http.ResponseWriter, clientReq
 	} else if targetResponse.ContentLength < 0 {
 		clientResponse.WriteHeader(targetResponse.StatusCode)
 		if _, err := io.CopyN(clientResponse, targetResponse.Body, handler.config.MaxBodySize); err != nil {
-			logger.Printf("Error relaying response body with unknown content-length: %s", err)
+			// NOTE: it is highly likely the server would come back without a content-length especially with
+			// mobile traffic. In this case, full copy happens but we get an EOF error that can be safely
+			// ignored. See this example: https://go.dev/play/p/8T3kbRfk_vO
```

Comment:
assigning @sethfowler for context on why this logs an error. Is there a cleaner way to run the copy here?

Reply:
My name probably appears on the blame because I've refactored this code a bit, but unfortunately I don't have full context. So this is just an educated guess, but...

I'd guess that this is just a bug. We should've distinguished between EOF (which is normal and reasonable here, since there may not be MaxBodySize bytes actually present) and other errors (which could indicate some kind of actual issue). I don't think there's any reason to log a message in the former case. It actually might make more sense to log a message if we _don't_ get EOF, since that'd indicate we're dropping some data on the floor.

I'd endorse updating this PR to log in those cases, if you have bandwidth, but the current version of the PR is already a clear improvement on the status quo, so I'll go ahead and approve it.

Reply:
Oh cool, that's something I wondered about. I'll make that update!